bin/check.sh

```diff
@@ -1,11 +1,20 @@
 #!/bin/bash
 set -ex
 
-buildifier -showlog -mode=check $(find . -type f \( -name 'BUILD' -or -name 'WORKSPACE' -or -wholename '.*bazel$' -or -wholename '.*bzl$' \) -print )
+bazel build @com_github_bazelbuild_buildtools//buildifier
+buildifier=$(bazel info bazel-bin)/external/com_github_bazelbuild_buildtools/buildifier/buildifier
+$buildifier -showlog -mode=check \
+  $(find . -type f \( -name 'BUILD' -or -name 'WORKSPACE' -or -wholename '.*bazel$' -or -wholename '.*bzl$' \) -print )
 
 NUM_CPU=$(getconf _NPROCESSORS_ONLN)
 
-gometalinter --concurrency=${NUM_CPU} --enable-gc --deadline=300s --disable-all\
+gometalinter="docker run \
+  -v $(bazel info output_base):$(bazel info output_base) \
+  -v $(pwd):/go/src/istio.io/pilot \
+  gcr.io/istio-testing/linter:bfcc1d6942136fd86eb6f1a6fb328de8398fbd80"
```

Comment:
The SHA corresponds to gometalinter SHA. We can automate Dockerfile.lint once we want to get monthly upstream updates.

Reply:
Will there be a permission issue here ? Will the link be created by root and not be readable by the user ?

Reply:
This image is supposed to be created once a month or even more rarely. Linters don't change that fast. The image should be public readable.

Reply:
So, this would require docker on the machine in order to even run the linter? That seems like a bummer :(

Reply:
But if it stabilizes the build infrastructure, so be it.

Reply:
i think we can change the script later to use system gometalinter. there's just that line at the end that needs to be changed to `./...` and use `-w` in docker.